knora/dsplib/utils/onto_create_ontology.py

```diff
@@ -148,73 +148,100 @@ def create_ontology(input_file: str,
+                                all_groups = Group.getAllGroups(con)
+
+                            # check that group exists
+                            for g in all_groups:
+                                if g.project.shortname == project_shortname and g.name == group_name:
+                                    group = g
+
+                        else:  # full_group_name refers to a group inside the same ontology
+                            group = new_groups.get(group_name)
+                            assert group is not None
+                        group_ids.add(group.id)
+                    elif len(tmp) == 1:
+                        if tmp[0] == "SystemAdmin":
+                            sysadmin = True
                     else:
-                        # we have ':groupname' and add to current project
-                        group = new_groups.get(tmp[1])
-                        assert group is not None
-                    group_ids.add(group.id)
-                else:
-                    if tmp[0] == "SystemAdmin":
-                        sysadmin = True
+                        print(f"ERROR Provided group name {full_group_name} for user {user.username} is not valid.")
 
             project_infos: Dict[str, bool] = {}
-            for projectname in user["projects"]:
-                # determine the project memberships of the user
-                # projectname has the form [projectname]:"member"|"admin" (projectname omitted = current project)
-                tmp = projectname.split(':')
-                assert len(tmp) == 2
-                if tmp[0]:
-                    # we have 'proj_shortname:"member"|"admin"'
-                    if not all_projects:
-                        all_projects = project.getAllProjects(con)
-                    tmp_project = list(filter(lambda g: g.shortname == tmp[0], all_projects))
-                    assert len(tmp_project) == 1
-                    in_project = tmp_project[0]
-                else:
-                    # we have ':"member"|"admin"'
-                    in_project = project
-                if tmp[1] == "admin":
-                    project_infos[in_project.id] = True
-                else:
-                    project_infos[in_project.id] = False
+
+            user_projects = user.get("projects")
+            # if "groups" is provided, add user to the group(s)
+            if user_projects:
+                for full_project_name in user_projects:
+                    if verbose:
+                        print(f"Add user to project: {full_project_name}")
+                    # full_project_name has the form [project_name]:member or [project_name]:admin
+                    # if project_name is omitted, the user is added to the current project
+                    tmp = full_project_name.split(':')
+
+                    if not len(tmp) == 2:
+                        print(f"ERROR Provided project name {full_project_name} for user {user.username} is not valid.")
```

Comment:
here you should probably add a `continue`, otherwise if `len(tmp)` is not 2, it would still attempt to do stuff, but probably crash or create nonsense

Reply:
Yes, very important indeed ;) Thanks for the hint, added a `continue`.